### DISCUSSION TITLE
How do you "call" a css file from your marp header

### DISCUSSION THREAD

**Discussion by gauthiervasseur** [marp-team/marp#536] [Q&A] [answered]
Greetings,

I can't the syntax to work to call a local CSS file from the marp header. Anyone has it?

**Comment by yhatt** [comment]
Marp's Markdown format _does not provide_ a specific syntax to call a local CSS file in Markdown front matter.

Instead, Marp always uses [a theme name](https://marpit.marp.app/directives?id=theme) to apply the theme CSS file.

```markdown
---
theme: default
---
```

If you have a custom theme CSS in the local, tell the location of your theme CSS into Marp tool you are using. For example, [Marp CLI](https://github.com/marp-team/marp-cli) can use [either `--theme` or `--theme-set` option](https://github.com/marp-team/marp-cli#theme), and [Marp for VS Code](https://github.com/marp-team/marp-vscode) can set ["Markdown: Marp > Themes" setting](https://github.com/marp-team/marp-vscode#use-custom-theme-css-%EF%B8%8F) (`markdown.marp.themes`). When using the custom theme CSS, please confirm that the theme name has been defined in the content of CSS as [the metadata `/* @theme xxx */`](https://marpit.marp.app/theme-css?id=metadata).

**Reply by gauthiervasseur** [reply]
Thank you, that is what I'm using right now. I pretty sure I made it work on a colleagues computer, using something like this:
![WhatsApp Image 2024-08-24 at 09 40 28](https://github.com/user-attachments/assets/ab1ba451-db58-4ae0-b60b-5e00b6f22516)

**Reply by yhatt** [reply]
For just calling a CSS file, [`@import` at-rule](https://developer.mozilla.org/docs/Web/CSS/@import) in the inline style is working, because Marp does not prevent CSS specific features. So your definition should actually achieve "call" a CSS file.

However, that is **not recommended within the context of Marp** because:

- The Marp converter **does not resolve** the content of imported styles through `@import`.
- So imported styles will **not apply Marp-specific CSS preprocessing**.
- This kind of styles might **disturb applying style correctly**, and might **break the interface of Marp-integrated apps**.

**Reply by gauthiervasseur** [reply]
Thank you so much. I'm trying to scale the distribution of styles across my team and tweaking VS code is not obvious for everyone!

**Reply by yhatt** [reply, marked as answer]
Related: marp-team/marpit#363

Marp team recommends to use `@import` carefully. It's no problem in Marp-specific usage (e.g. [importing other theme](https://marpit.marp.app/theme-css?id=import-rule)) or known usage (e.g. [Google Fonts](https://fonts.google.com/), whose only `@font-face` definitions in the most case), but the other usage may met the some problem.

**Reply by gauthiervasseur** [reply]
Thank you!
